### PR TITLE
feat(sass): provide central entrypoint

### PR DESCRIPTION
**Pull request by tay1orjones**
**Summary**

- add an `index.scss` for use in consuming projects

**Change List (commits, features, bugs, etc)**

- modify import paths in styles.scss
- add .scss files to import component styles so we're not copying as many files from `node_modules` on build
- modify rollup build to copy appropriate files to globals, components, and styles.scss
  - added a rename helper function to assist with this
- add new step to build scrips to `rm -rf` remove generated directories before a new build

**Related Issues**

- #426 

**Acceptance Test (how to verify the PR)**

- pull down the branch, run `yarn`, view the `./lib` folder
- please double check the new component scss files for proper import paths. 
  - ex: `_accordion.scss` actually imports `@import '~carbon-components/scss/components/accordion/accordion';` and not some other component by accident.
- Are we missing anything?

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -55,102 +67,36 @@ export default {
...
-            'node_modules/carbon-components/scss/components/search',
-            'node_modules/carbon-components/scss/components/select',
-            'node_modules/carbon-components/scss/components/text-input',
-            'node_modules/carbon-components/scss/components/text-area',
-            'node_modules/carbon-components/scss/components/number-input',
-            'node_modules/carbon-components/scss/components/form',
-            'node_modules/carbon-components/scss/components/link',
             'node_modules/carbon-components/scss/components/list-box',
             'node_modules/carbon-components/scss/components/list',
             'node_modules/carbon-components/scss/components/data-table',
```
Comment (tay1orjones) [inline rollup.config.js]: @davidicus There are 5 component files we're still directly copying and importing from `node_modules`

1. `'node_modules/carbon-components/scss/components/list-box'`
1. `'node_modules/carbon-components/scss/components/list'`
1. `'node_modules/carbon-components/scss/components/data-table'`
1. `'node_modules/carbon-components/scss/components/skeleton'`
1. `'node_modules/carbon-components/scss/components/pagination-nav'`

I'm not sure what to do with these. They're either not exported directly from carbon-components-react, or we have our own version (data-table). I could create a folder and file for them like I have the others, but as we're not exporting them, this feels odd. Thoughts?

File: src/styles.scss [hunk 2]
```diff
@@ -108,53 +108,52 @@ $deprecations--message: 'Deprecated code was found, this code will be removed be
...
+@import 'components/Pagination/pagination';
+@import 'components/Accordion/accordion';
+@import 'components/Breadcrumb/breadcrumb';
+@import 'components/Toolbar/toolbar';
+@import 'components/TimePicker/time-picker';
+@import 'components/Slider/slider';
+@import 'components/Tile/tile';
+@import '~carbon-components/scss/components/skeleton/skeleton'; // ???
+@import 'components/InlineLoading/inline-loading';
+@import '~carbon-components/scss/components/pagination-nav/pagination-nav'; // ???
```
Comment (tay1orjones) [inline src/styles.scss]: I placed `// ???` by them to denote which ones I'm unsure of here.

File: rollup.config.js (hunk 1 above)
Comment (davidicus) [reply rollup.config.js]: I think its probably a good idea to create folder and file for each and import them there as its consistent and if we ever do export them as modules they will have the infrastructure in place. The data-table partial can probably be brought into our table component no? I know we export both carbon tables and our own but the styles on top of these shouldnt clash

Comment (tay1orjones) [reply rollup.config.js]: 👍 Will add folders for these so we're not copying directly from `node_modules`

**Comment by tay1orjones** [comment]
@davidicus ready for review

**Pull request merged**